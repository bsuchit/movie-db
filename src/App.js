import React, { useEffect } from 'react';
import PropTypes from 'prop-types';
import { Image, Loader } from 'semantic-ui-react';

import MovieList from 'components/list/MovieList';
import Logo from 'resources/star-wars-logo.png';
function App({ onMount, movies = [], isLoading = false, error = false}) {
  useEffect(() => {
    onMount && onMount();
  }, [onMount]);

  if (error) {
    return <label>Error Occurred</label>
  }

  return (
    <div>
      <Image src={Logo} className="logo" />
      <div>
        {
          isLoading ? 
          <Loader active>Loading...</Loader>
          :
          <MovieList
            movies={movies}
          /> 
        }
      </div>
    </div>
  );
}

App.propTypes = {
  onMount: PropTypes.func,
  movies: PropTypes.array,
  isLoading: PropTypes.bool,
  error: PropTypes.bool
};

export default App;
