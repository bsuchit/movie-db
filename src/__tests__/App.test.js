import React from 'react';
import { render, screen } from '@testing-library/react';
import App from '../App';

describe('Given App.js', () => {
  it('should render app.js', () => {
    const movies = [{
      title: 'A New Hope',
      release_date: '1980-12-05'
    },
    {
      title: 'Return of the jedi',
      release_date: '1983-05-25'
    }];

    const mockOnMount = jest.fn();

    const { container } = render(<App onMount={mockOnMount} movies={movies} isLoading={false} error={false} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should show loader WHEN loading', () => {
    const mockOnMount = jest.fn();

    const { container } = render(<App onMount={mockOnMount} movies={[]} isLoading={true} error={false} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should show error WHEN failed', () => {
    const mockOnMount = jest.fn();

    const { container } = render(<App onMount={mockOnMount} movies={[]} isLoading={false} error={true} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders Sort By button', () => {
    render(<App />);
    const linkElement = screen.getByText(/Sort By/i);
    expect(linkElement).toBeInTheDocument();
  });
});
