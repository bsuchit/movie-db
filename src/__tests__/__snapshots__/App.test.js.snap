// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given App.js should render app.js 1`] = `
<div>
  <img
    class="ui image logo"
    src="star-wars-logo.png"
  />
  <div>
    <div>
      <div>
        <div
          class="row list_header"
        >
          <button
            class="ui button sortby_button"
          >
            Sort By
          </button>
          <div
            class="ui icon input search_input"
            data-testid="search"
          >
            <input
              placeholder="Type to search..."
              type="text"
              value=""
            />
            <i
              aria-hidden="true"
              class="search icon"
            />
          </div>
        </div>
      </div>
      <div
        class="list_container"
      >
        <div
          class="list"
        >
          <div
            class="movie_card"
          >
            <div
              class="info_section"
              data-testid="A New Hope"
            >
              <div
                class="movie_header"
              >
                <div
                  class="column"
                >
                  <h1>
                    A New Hope
                  </h1>
                  <p
                    class="type"
                  >
                    Episode undefined
                  </p>
                </div>
                <div
                  class="vertical"
                />
                <div
                  class="column"
                >
                  <h4>
                    Release Date: 1980-12-05
                  </h4>
                  <h4>
                    Director: undefined
                  </h4>
                </div>
              </div>
            </div>
          </div>
          <div
            class="movie_card"
          >
            <div
              class="info_section"
              data-testid="Return of the jedi"
            >
              <div
                class="movie_header"
              >
                <div
                  class="column"
                >
                  <h1>
                    Return of the jedi
                  </h1>
                  <p
                    class="type"
                  >
                    Episode undefined
                  </p>
                </div>
                <div
                  class="vertical"
                />
                <div
                  class="column"
                >
                  <h4>
                    Release Date: 1983-05-25
                  </h4>
                  <h4>
                    Director: undefined
                  </h4>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="auto_margin"
        >
          Please select movie to see preview.
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Given App.js should show error WHEN failed 1`] = `
<label>
  Error Occurred
</label>
`;

exports[`Given App.js should show loader WHEN loading 1`] = `
<div>
  <img
    class="ui image logo"
    src="star-wars-logo.png"
  />
  <div>
    <div
      class="ui active text loader"
    >
      Loading...
    </div>
  </div>
</div>
`;
